### PR TITLE
Implement dot operator

### LINKED ISSUE #52: Dot operator to repeat last action
# Goal

As in vim, we'd like to implement a dot operator, bound to `.` This would cause the previous action to repeat. For example if the last action was `10j`, then `.` would repeat this action.

# Implementation

* This is probably just a matter of storing an `Action` enum in the ui `State`, and on a `.` keypress, dispatching the last `Action` variant.
* It will also mean updating that last action on all other keypresses.
* The default for the last action, used in the `init` func should be Noop.
* There may be some actions which are non-repeatable, and as such the last action should be updated to be `Noop`.

### PR DESCRIPTION
**Pull request by Davejkane**
This closes [#52](https://github.com/Davejkane/riv/issues/52).
* Add the last action to ui state
* Processes the period key
* process all actions returned from keyhandling and clone into last_action and return the action.

### CODE REVIEW

File: src/ui.rs [hunk 1]
```diff
@@ -54,6 +55,24 @@ pub struct State {
...
     pub fullscreen: bool,
+    /// last_action records the last action performed. Used for repeating that action
+    pub last_action: Action,
+}
+
+impl State {
+    /// update_last_action takes an action, sets the last_action to said action, and returns the Action
+    fn process_action(&mut self, a: Action) -> Action {
+        match a {
+            Action::Quit | Action::ToggleFullscreen | Action::ReRender => {
```
Comment (gurgalex) [inline src/ui.rs]: I assume `ToggleFullscreen` is here because it needs to rerender the screen?

**Review by gurgalex — COMMENTED** [review]
Looks fine

**Comment by gurgalex** [comment]
Sorry about the merge conflict from #54

**Review by gurgalex — CHANGES_REQUESTED** [review]
Great!

Could you exempt `reRendering` from being or resetting `last_state`?

That way `ToggleFullscreen` can be switched between with `.` as well.

**Review by gurgalex — APPROVED** [review]

**Comment by gurgalex** [comment]
My solution may have been hacky. `ReRender` is never set as `last_action`.

```rust
        match a {
            Action::Quit => {
                self.last_action = Action::Noop;
                a
            }
            Action::ReRender => a,
            _ => {
                self.last_action = a.clone();
                a
            }
        }

```

**Comment by Davejkane** [comment]
I'm not sure I understand. Are you suggesting that I do what you posted in that last code sample? What does that change?

**Comment by gurgalex** [comment]
On fullscreen switch multiple events which emit `ReRender` can be fired as well.
`Resized`, `SizeChanged`, `Moved`, `exposed`

So, the `last action` is set to 

`ToggleFullscreen` and then `Noop` immediately after.

Special casing Rerender to not touch `last_action` preserves any action prior to the action `ReRender`.

**Comment by gurgalex** [comment]
Primary internet access is out in my area. So can you merge this?

**Pull request merged**